### PR TITLE
Robust checking subject alternative name

### PR DESCRIPTION
**Pull request by RichardChen820**
Robust checking subject alternative name

### CODE REVIEW

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/DomainVerifier.cs [hunk 1]
```diff
@@ -66,36 +73,40 @@ private static IEnumerable<string> GetDomainsFromSanExtension(X509Certificate2 c
 
             X509Extension sanExtension = cert.Extensions[SubjectAltNameOid];
 
-            if (sanExtension != null)
+            // For the DNS name, the tag class is ContextSpecific and tag value is 2 according to RFC 5280
+            var dnsNameTag = new Asn1Tag(TagClass.ContextSpecific, tagValue: 2, isConstructed: false);
+
+            AsnReader reader = new AsnReader(sanExtension.RawData, AsnEncodingRules.BER);
```
Comment (jimmyca15) [inline src/Microsoft.Extensions.Configuration.AzureAppConfiguration/DomainVerifier.cs:79]: Is any of this disposable?

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/DomainVerifier.cs [hunk 2]
```diff
@@ -19,7 +20,6 @@ internal static class DomainVerifier
     {
         private const int TlsPort = 443;
         private const string SubjectAltNameOid = "2.5.29.17";
-        private const string DnsName = "DNS Name=";
 
         public static async Task<IEnumerable<string>> GetValidDomains(Uri originEndpoint, string srvHostName)
```
Comment (jimmyca15) [inline src/Microsoft.Extensions.Configuration.AzureAppConfiguration/DomainVerifier.cs:24]: I think we need to add a test for this that calls a fake App Config endpoint like azure.aconfig.io and tries to get the domains.

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/DomainVerifier.cs (hunk 1 above)
Comment (RichardChen820) [reply src/Microsoft.Extensions.Configuration.AzureAppConfiguration/DomainVerifier.cs:79]: I had checked,  it's not disposable

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/DomainVerifier.cs (hunk 2 above)
Comment (RichardChen820) [reply src/Microsoft.Extensions.Configuration.AzureAppConfiguration/DomainVerifier.cs:24]: Then we expect it fails, get no valid domain?

**Pull request closed without merging**